### DISCUSSION TITLE
rocsys and rocprofv2

### DISCUSSION THREAD

**Discussion by claudiubalogh** [ROCm/ROCm#3945] [Q&A] [answered]
Hello all

I am trying to profile my very long running OpenCL app with:

```rocsys --session rocmmmm launch rocprofv2  --hsa-trace --kernel-trace -d /root/rocprof/ --plugin perfetto```

rocsys start and stop seem to be working ok, writing tens of MB of data every few seconds once started.
However, after I send the stop command rocprofv2 seems to constantly write little amounts of data (under 1kB every few seconds) in the destination directory. I left it running yesterday after sending the stop command and it indeed wrote all night long, I don't know what, because... and here comes the more stringent problem:
The output data seems to be corrupted after less than one minute of data collection (perfetto reports "data losses"). Depending on the job I send to the app it can run from a few seconds to tens of minutes. Now I don't need tens of minutes of profiling data, but at least a few minutes would be useful.

My question is are these tools ready for prime time? What is your experience?

**Comment by sohaibnd** [comment]
Hi @claudiubalogh, can you share a screenshot of what you see on perfetto when trying to open the trace file?

**Comment by sohaibnd** [comment, marked as answer]
@claudiubalogh I suggest using the new rocprofv3 built on top of [rocprofiler-sdk](https://rocm.docs.amd.com/projects/rocprofiler-sdk/en/latest/index.html) since rocprofv2 is no longer under development (it never made it out of the beta stage actually). To control profiling scope, rocprofv3 provides the [roctxProfilerPause](https://github.com/ROCm/rocprofiler-sdk/blob/9de284a56886f09484b0b7e3d281bab834d3de73/source/include/rocprofiler-sdk-roctx/roctx.h#L139) and [roctxProfilerResume](https://github.com/ROCm/rocprofiler-sdk/blob/9de284a56886f09484b0b7e3d281bab834d3de73/source/include/rocprofiler-sdk-roctx/roctx.h#L157) functions (these will be added to the docs soon). I also haven't observed any data losses being reported with rocprofv3.

I'm marking this response as the answer, but if you run into any issues with rocprofv3, feel free to post them here or create an issue under the [rocprofiler-sdk](https://github.com/ROCm/rocprofiler-sdk) and we can have a look.
